src/lib.rs

```diff
@@ -206,7 +206,14 @@ pub struct Peripherals {
     pub PMU: PMU,
 
     /// Switch matrix
-    pub SWM: SWM,
+    ///
+    /// The Switch matrix is enabled by default.
+    #[cfg(feature = "82x")]
+    pub SWM: SWM<init_state::Enabled>,
+
+    /// Switch matrix
+    #[cfg(feature = "845")]
+    pub SWM: SWM<init_state::Disabled>,
```

Comment:
I think the documentation is not ideal. Only one version will show up in online documentation, and that is bound to confuse users of the other variant.

Suggestion: Explain the whole situation (default-enabled on LPC82x, not on LPC845), copy-paste the documentation to both variants, and try to keep that in sync forever.

Not ideal, but probably good enough. And I can't think of a better solution (code generation would probably be overkill).